src/main/java/rx/internal/operators/OperatorGroupBy.java

```diff
@@ -132,6 +151,34 @@ public GroupBySubscriber(Subscriber<? super GroupedObservable<K, V>> actual, Fun
             this.requested = new AtomicLong();
             this.groupCount = new AtomicInteger(1);
             this.wip = new AtomicInteger();
+            if (mapFactory == null) {
+                this.groups = new ConcurrentHashMap<Object, GroupedUnicast<K, V>>();
+                this.evictedKeys = null;
+            } else {
+                this.evictedKeys = new ConcurrentLinkedQueue<K>();
+                this.groups = createMap(mapFactory, new EvictionAction<K>(evictedKeys));
+            }
+        }
+        
+        //declare a class instead of using anonymous class to 
+        //limit enclosing scope. Is this of value?
```

Comment:
yup + makes stactraces better, @akarnokd recently made similar change and coverted a lot of anonymous classes into static inner ones

I'd remove comment to this class